### PR TITLE
clarify lexprs/rexprs

### PR DESCRIPTION
**Pull request by arielb1**

### CODE REVIEW

File: specification/terms.md [hunk 1]
```diff
@@ -70,28 +72,29 @@ A typed expression that evaluates to an object, as opposed to a value. The LHS o
...
-If a lexpr is used where an vexpr is expected, the memory is read from the object referred to by the lexpr to create a vexpr *if* either T: Copy, or the mutability of the lexpr allows for moving.
-
-### §2.3 Temporary lexpr
+NOTE: Because of evaluation order concerns, after being potentially wrapped to a
+vexpr, subexpressions are often wrapped in a `VExpr::Use(LExpr::Temp(vexpr))`. The
+exact rules here should be documented.
 
-An implicit lexpr that wraps a vexpr when an lexpr is expected.
+If a lexpr is used where an vexpr is expected, it is wrapped in an implicit use
+vexpr, the rules of which are given in §2.3.
```
Comment (strega-nil) [inline specification/terms.md:84]: This feels like an overcomplicated way of saying this.

File: specification/terms.md [hunk 2]
```diff
@@ -1,5 +1,7 @@
 Some terms, defined by @ubsan and @arielb1. Please debate on them, or correct them, as you see fit :)
 
+For explicitness, some of the section contain ADT declarations. These ADT declarations use Rust syntax for familiarity, but all smart pointers have been removed for clarity.
+
```
Comment (strega-nil) [inline specification/terms.md:4]: I don't think this is necessary.

File: specification/terms.md [hunk 3]
```diff
@@ -106,14 +109,51 @@ fn main() {
...
 ```
 
-### §2.4 Note on the terms lvalue and rvalue
+### §2.4 Use vexpr
+
+```Rust
+    VExpr::Use(LExpr)
+```
+
+An implicit vexpr that wraps a lexpr when a vexpr is expected.
```
Comment (strega-nil) [inline specification/terms.md:118]: This also seems overcomplicated in order to fit the ADT model.

**Comment by strega-nil** [comment]
Otherwise, sgtm.

**Comment by arielb1** [comment]
C has reified lvalue-to-rvalue conversions, so why not us?

**Comment by strega-nil** [comment]
@arielb1 I don't mind lvalue-rvalue conversions, in fact I like them, I just don't like the "use" terminology. "Wrapping in a use" sounds really weird.

File: specification/terms.md [hunk 4]
```diff
@@ -220,3 +220,33 @@ A value, in order to get stored into memory, must first undergo the process of t
 ### §4.4 Conversion from Memory
 
 Conversion from memory will be a conversion to memory, but in reverse, going from computer-friendly form to pure mathematical object.
+
+## §5 Memory
+
+### §5.1 Segmentation
```
Comment (strega-nil) [inline specification/terms.md:230]: This seems really unnecessary.

File: specification/terms.md (hunk 3 above)
Comment (arielb1) [reply specification/terms.md:118]: But my mental model is an ADT model and I want to keep it. For prior art, the C standard _also_ has lvalue-to-rvalue conversions.

**Comment by arielb1** [comment]
Removed the segmentation bit.

**Comment by arielb1** [comment]
This was opened on the wrong branch

**Pull request closed without merging**